src/magic_folder/snapshot.py

```diff
@@ -0,0 +1,247 @@
+        name = "default"
+    nodedir = config.get_config_path()
+    magic_folders = load_magic_folders(nodedir)
+    if name not in magic_folders:
+        raise RuntimeError(
+            "No magic-folder named '{}'".format(name)
+        )
+
+    # we will always have authorship information for this
+    # magic-folder; "legacy" magic-folders will go through "tahoe
+    # migrate" first and have an author created.
+
+    author_raw = config.get_private_config("magicfolder_{}.privkey".format(name))
+    author_data = json.loads(author_raw)
+
+    return LocalAuthor(
+        name=author_data[u"author_name"],
+        signing_key=SigningKey(
+            author_data[u"author_private_key"],
+            encoder=Base64Encoder,
+        ),
+    )
+
+
+@attr.s
+class LocalSnapshot(object):
+    name = attr.ib()
+    author = attr.ib()
+    metadata = attr.ib()
+    content_path = attr.ib()  # full filesystem path to our stashed contents
+    parents_local = attr.ib()  # LocalSnapshot instances
+
+    # once we do uploads / downloads and have RemoteSnapshots, we will
+    # also have those kind of parents too:
+    # parents_remote = attr.ib()
+    # ..and need to add methods here to count and async get parents
+
+    def get_content_producer(self):
+        """
+        :returns: an IBodyProducer that gives you all the bytes of the
+            on-disc content. Raises an error if we already have a
+            capability. Note that this data will have been stashed previously.
+        """
+        return FileBodyProducer(
+            open(self.content_path, "rb")
+        )
+
+    def _get_synchronous_content(self):
+        """
+        For testing only.
+        :returns: the content immediately
+        """
+        with open(self.content_path, "rb") as f:
+            return f.read()
+
+
+@inlineCallbacks
+def create_snapshot(name, author, data_producer, snapshot_stash_dir, parents=None):
+    """
+    Creates a new LocalSnapshot instance that is in-memory only. All
+    data is stashed in `snapshot_stash_dir` before this function
+    returns.
+
+    :param author: LocalAuthor instance (which will have a valid
+        signing-key)
+
+    :param data_producer: file-like object that can deliver all the
+        data for the content of this Snapshot (it will be read
+        immediately).
+
+    :param snapshot_stash_dir: the directory where Snapshot contents
+        are to be stashed.
+
+    :param parents: a list of LocalSnapshot instances (may be empty,
+        which is the default if not specified).
+    """
+    if parents is None:
+        parents = []
+
+    if not isinstance(author, LocalAuthor):
+        raise ValueError(
+            "create_snapshot 'author' must be a LocalAuthor instance"
+        )
+
+    # when we do uploads, we will distinguish between remote and local
+    # parents, so the "parents" list may contain either kind in the
+    # future.
+    parents_local = []
+    for idx, parent in enumerate(parents_local):
```

Comment:
`enumerate(parents)` ? Because `parents` is a list of only LocalSnapshots at the moment?